go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -1203,6 +1220,285 @@ func (ts *trafficSwitcher) isSequenceParticipating(ctx context.Context) (bool, e
+		if kerr != nil {
+			return vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get vschema for keyspace %s: %v",
+				keyspace, kerr)
+		}
+		if kvs == nil || kvs.Sharded || len(kvs.Tables) == 0 {
+			return nil
+		}
+		for tableName, tableDef := range kvs.Tables {
+			select {
+			case <-sctx.Done():
+				return sctx.Err()
+			case <-searchCompleted:
+				return nil
+			default:
+			}
+			smMu.Lock() // Prevent concurrent access to the map
+			sm := sequencesByBackingTable[tableName]
+			if tableDef != nil && tableDef.Type == vindexes.TypeSequence &&
+				sm != nil && tableName == sm.backingTableName {
+				tablesFound++ // This is also protected by the mutex
+				sm.backingTableKeyspace = keyspace
+				sm.backingTableDBName = "vt_" + keyspace
+				if tablesFound == tableCount { // Short circuit the search
+					smMu.Unlock()
+					select {
+					case <-searchCompleted: // It's already been closed
+						return nil
+					default:
+						close(searchCompleted) // Mark the search as completed
+						return nil
+					}
+				}
+			}
+			smMu.Unlock()
+		}
+		return nil
+	}
+	keyspaces, err := ts.TopoServer().GetKeyspaces(ctx)
+	if err != nil {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to get keyspaces: %v", err)
+	}
+	searchGroup, gctx := errgroup.WithContext(ctx)
+	for _, keyspace := range keyspaces {
+		keyspace := keyspace // https://golang.org/doc/faq#closures_and_goroutines
+		searchGroup.Go(func() error {
+			return searchKeyspace(gctx, keyspace)
+		})
+	}
+	if err := searchGroup.Wait(); err != nil {
+		return nil, err
+	}
+
+	if tablesFound != tableCount {
+		return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "failed to locate all of the backing sequence tables being used; sequence table metadata: %+v",
+			sequencesByBackingTable)
+	}
+	return sequencesByBackingTable, nil
+}
+
+// findSequenceUsageInKeyspace searches the keyspace's vschema for usage
+// of sequences. It returns a map of sequence metadata keyed by the backing
+// sequence table name -- if any usage is found -- along with a boolean to
+// indicate if all of the backing sequence tables were defined using
+// qualified table names (so we know where they all live) along with an
+// error if any is seen.
+func (ts *trafficSwitcher) findSequenceUsageInKeyspace(ctx context.Context, vschema *vschemapb.Keyspace) (map[string]*sequenceMetadata, bool, error) {
+	allFullyQualified := true
+	targets := maps.Values(ts.Targets())
+	if len(targets) == 0 || targets[0].GetPrimary() == nil { // This should never happen
+		return nil, false, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "no primary tablet found for target keyspace %s", ts.targetKeyspace)
+	}
+	targetDBName := targets[0].GetPrimary().DbName()
+	sequencesByBackingTable := make(map[string]*sequenceMetadata)
+
+	for _, table := range ts.Tables() {
+		vs, ok := vschema.Tables[table]
+		if !ok || vs == nil {
+			continue
+		}
+		if vs.AutoIncrement != nil && vs.AutoIncrement.Sequence != "" {
```

Comment:
just a suggestion: negate the condition and do an early `continue`, like you did above.

Reply:
Good call. Done: https://github.com/vitessio/vitess/pull/13656/commits/960056658d0b474d6dca591c6307aab3f7ab9197